datajunction-server/datajunction_server/api/measures.py

```diff
@@ -0,0 +1,145 @@
+"""
+Measures related APIs.
+"""
+
+import logging
+from typing import List
+
+from fastapi import Depends
+from sqlmodel import Session, select
+
+from datajunction_server.api.helpers import get_node_by_name
+from datajunction_server.errors import DJAlreadyExistsException, DJDoesNotExistException
+from datajunction_server.internal.authentication.http import SecureAPIRouter
+from datajunction_server.models import Column
+from datajunction_server.models.measure import (
+    CreateMeasure,
+    EditMeasure,
+    Measure,
+    MeasureOutput,
+    NodeColumn,
+)
+from datajunction_server.utils import get_session, get_settings
+
+_logger = logging.getLogger(__name__)
+settings = get_settings()
+router = SecureAPIRouter(tags=["measures"])
+
+
+def get_measure_by_name(
+    session: Session,
+    measure_name: str,
+    raise_if_not_exists: bool = True,
+) -> Measure:
+    """Retrieve a measure by name"""
+    measure = (
+        session.exec(select(Measure).where(Measure.name == measure_name))
+        .unique()
+        .one_or_none()
+    )
+    if raise_if_not_exists and not measure:
+        raise DJDoesNotExistException(
+            message=f"Measure with name `{measure_name}` does not exist",
+        )
+    return measure
+
+
+def get_node_columns(session: Session, node_columns: List[NodeColumn]) -> List[Column]:
+    """
+    Finds all the specified node columns or raises if they don't exist
+    """
+    measure_columns = []
+    for node_column in node_columns:
+        node = get_node_by_name(session, node_column.node)
+        available = [
+            col for col in node.current.columns if col.name == node_column.column
+        ]
+        if len(available) != 1:
+            raise DJDoesNotExistException(
+                message=f"Column `{node_column.column}` does not exist on "
+                f"node `{node_column.node}`",
+            )
+        measure_columns.extend(available)
+    return measure_columns
+
+
+@router.get("/measures/", response_model=List[str])
+def list_measures(*, session: Session = Depends(get_session)) -> List[str]:
```

Comment:
We could do add a `prefix` argument here and have this case be server when that arg is empty?